azure-pipelines.yml

```diff
@@ -0,0 +1,416 @@
+variables:
+  buildNumber: $[ counter('constant', 13000) ] # Start higher than our AppVeyor versions. Every build (pr or branch) will increment.
+
+name: 2.0.$(buildNumber)
+  
+pr:
+  branches:
+    include:
+    - master
+    - dev
+
+trigger:
+  branches:
+    include:
+    - master
+    - dev
+
+jobs:
+- job: InitializePipeline
+  pool:
+    vmImage: 'vs2017-win2016'
+  steps:
+  - task: AzureKeyVault@1
+    inputs:
+      # Note: This is actually a Service Connection in DevOps, not an Azure subscription name
+      azureSubscription: 'Azure-Functions-Host-CI'
+      keyVaultName: 'azure-functions-host-ci'
+      secretsFilter: '*'
+
+- job: BuildArtifacts_Ubuntu
+  pool:
+    vmImage: 'ubuntu-18.04'
+  steps:
+  - task: Bash@3
+    inputs:
+      targetType: 'inline'
+      script: |
+        mkdir .dotnet &&
+        chmod +x dotnet-install.sh &&
+        ./dotnet-install.sh --version 2.2.202 --install-dir .dotnet &&
+        PATH=".dotnet:"$PATH && dotnet --info
+        dotnet build WebJobs.Script.sln
+
+- job: BuildArtifacts_Windows
+  dependsOn: InitializePipeline
+  condition: succeeded()
+  pool:
+    vmImage: 'vs2017-win2016' 
+  steps:
+  - task: UseDotNet@2
+    inputs:
+      packageType: 'sdk'
+      version: '2.2.x'
+      performMultiLevelLookup: true
+  - task: PowerShell@2
+    displayName: "Determine suffix and populate PR title"
+    inputs:
+      targetType: 'inline'
+      script: |
+        $sourceBranch = $env:BUILD_SOURCEBRANCH
+        $buildReason = $env:BUILD_REASON
+        $title = ""
+        Write-Host "SourceBranch: $sourceBranch, Build reason: $buildReason"
+        if($sourceBranch.endsWith('master') -and ($buildReason -ne "PullRequest"))
+        {
+          "##vso[task.setvariable variable=IncludeSuffix]0" #Setting to false
+        }
+        else
+        {
+          "##vso[task.setvariable variable=IncludeSuffix]1"
+          if($buildReason -eq "PullRequest")
+          {
+            $response = Invoke-RestMethod api.github.com/repos/$env:BUILD_REPOSITORY_ID/pulls/$env:SYSTEM_PULLREQUEST_PULLREQUESTNUMBER
+            $title = $response.title.ToLowerInvariant()
+          }
+        }
+        "##vso[task.setvariable variable=PullRequestTitle]$title"
```

Comment:
I'd like to avoid putting too much logic into these yml files. Can you do all of this in `build/initialize-pipeline.ps1`? The idea in v3.x was that this would run in `InitializePipeline` before `BuildArtifacts` and set up all the env vars to determine whether we could skip the entire build job. 

Then you don't need all of the repeated conditions below either -- the condition goes on the entire BuildArtifiacts job.

Reply:
I initially had trouble referencing the variables from the other job, then realized I was referencing it wrong. Just updated it.